statemachine/statemachine.py

```diff
@@ -289,6 +306,9 @@ def check(self):
             raise InvalidDefinition(_('There should be one and only one initial state. '
                                       'Your currently have these: {!r}'.format(initials)))
         self.initial_state = initials[0]
+        if (not self._is_connected(self.initial_state)):
```

Comment:
The method can be something like `self._not_connected_states(state)`, returning `None` or a list of unreachable states. This way we can tell the user on the exception.